src/WarcraftLegacies.Source/Quests/Goblin/QuestRatchet.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using MacroTools.ControlPointSystem;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.LegendSystem;
+using MacroTools.ObjectiveSystem.Objectives.ControlPointBased;
+using MacroTools.ObjectiveSystem.Objectives.FactionBased;
+using MacroTools.ObjectiveSystem.Objectives.LegendBased;
+using MacroTools.ObjectiveSystem.Objectives.UnitBased;
+using MacroTools.QuestSystem;
+using static War3Api.Common;
+
+
+namespace WarcraftLegacies.Source.Quests.Goblin
+{
+  /// <summary>
+  /// The Goblins can acquire Ratchet.
+  /// </summary>
+  public sealed class QuestRatchet : QuestData
+  {
+    private readonly List<unit> _rescueUnits;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QuestRatchet"/>.
+    /// </summary>
+    /// <param name="noggenfogger">Must be brought somewhere to complete the quest.</param>
+    public QuestRatchet(LegendaryHero noggenfogger) : base("Ratchet Port",
+      "The port of Ratchet would be a great expansion of our trade empire.",
+      @"ReplaceableTextures\CommandButtons\BTNGoblinShop2.blp")
+    {
+      AddObjective(new ObjectiveControlPoint(ControlPointManager.Instance.GetFromUnitType(Constants.UNIT_N0A6_RATCHET)));
+      AddObjective(new ObjectiveLegendInRect(noggenfogger, Regions.Ratchet_Unlock_1, "Ratchet"));
+      AddObjective(new ObjectiveSelfExists());
+      Required = true;
+      ResearchId = Constants.UPGRADE_VQ01_QUEST_COMPLETED_RATCHET_PORT;
+      _rescueUnits = Regions.Ratchet_Unlock_1.PrepareUnitsForRescue(RescuePreparationMode.HideNonStructures,
+        filterUnit => filterUnit.GetTypeId() != FourCC("ngme"));
+    }
+
+    /// <inheritdoc />
+    protected override string RewardFlavour => "With a port now established near Orgrimmar, we now have an outpost to support the Horde war effort in the North!";
+
+    /// <inheritdoc />
+    protected override string RewardDescription => "You gain control of a small outpost in Ratchet";
```

Comment:
No need for the "you", since it's implied in RewardDescription. I would suggest:
```suggestion
    protected override string RewardDescription => "Gain control of all units in Ratchet";
```

Reply:
Makes sense. Theramore also doesn't implicitly reference "You" but with the reference to the faction heroes (in this case Gazlowe) I think the text "Gain control.." is sufficient

Reply:
Not out of date, I'm just dumb and though implicit was explicit 😆